force-app/main/default/classes/BDI_DataImportCTRL_TEST.cls

```diff
@@ -162,6 +160,172 @@ private with sharing class BDI_DataImportCTRL_TEST {
         System.assertEquals(label.bdiDryRunNoMatch, listDI[1].Contact2ImportStatus__c);
     }
 
+    /*********************************************************************************************************
+    * @description Tests that controller will properly run dry run for only the selected DI records
+    */
+    static testMethod void testSelectedDIDryRun() {
+
+        // Set the contact matching rule so that matching during dry run should happen
+        Data_Import_Settings__c dis = UTIL_CustomSettingsFacade.getDataImportSettings();
+        dis.Contact_Matching_Rule__c = 'Firstname,Lastname,Email';
+        UTIL_CustomSettingsFacade.setDataImportSettings(dis);
+
+        // This contact will cause the Data Import record to match on it
+        Contact testContA = new Contact(FirstName = 'Susie',
+                                         LastName = 'TestGroupA01',
+                                        npe01__HomeEmail__c = 'testgroupAcontact01Personal@fakedata.com',
+                                        Email = 'testgroupAcontact01Personal@fakedata.com');
+        insert testContA;
+
+        // This DI will the be passed into the controller and processed
+        DataImport__c testDataImportA = new DataImport__c(
```

Comment:
Would it make sense to put Data Import record creation into a test util class of some sort?

Reply:
I think thats a good idea, but I think I would like to hold off at this time.